indra/llwindow/llkeyboard.cpp

```diff
@@ -272,6 +282,11 @@ BOOL LLKeyboard::handleTranslatedKeyUp(KEY translated_key, U32 translated_mask)
 		// sequence W<return> in chat to move agents forward. JC
 		mKeyUp[translated_key] = TRUE;
 		handled = mCallbacks->handleTranslatedKeyUp(translated_key, translated_mask);
+        // TODO: capture unhandled keys and translate any mapped buttons to game controller
```

Comment:
if you want to translate keyboard buttons (our mouse buttons) to game control buttons according to user's settings, it should be LLViewerInput.

Reply:
Thanks for the input.  Yes, we will add the ability to map keys to game controller input and when we get it working it will be moved to a more correct code context.  This is a work in progress (WIP).  We might add mouse mappings although I'm less certain how that would work right now so am punting on it.

Reply:
>We might add mouse mappings although I'm less certain how that would work right now so am punting on it.

At the moment viewer treats mouse keys the same way as keyboard. There is a bunch of trigerable functions, we map keyboard or mouse keys to those functions based on user settings (be those purely local like turning on voice, or server handled 'click' for LLTakeControl) to be trigered on press and release. Ideally any joystick specific keys should work the same way and just be hooked into LLViewerInput with LLViewerInput triggering assigned action.